sdk/iothub/azure-iot-deviceprovisioning/README.md

```diff
@@ -0,0 +1,283 @@
+client.individual_enrollments.create_or_update(
+    id="<enrollment_id>",
+    enrollment = {
+        "registrationId": "<enrollment_id>"
+        "attestation": {
+            "type": "symmetricKey",
+        },
+        "deviceId": "<device_id>",
+        "reprovision_policy": reprovision_policy
+    }
+)
+```
+
+### Create an intermediate x509 certificate enrollment group
+Create an x509 enrollment group to provision one or more devices using x509 attestation.
+```python
+from azure.iot.deviceprovisioning import DeviceProvisioningClient
+
+# Initialize client
+client = DeviceProvisioningClient.from_connection_string(connection_string="<connection_string>")
+
+# Load certificate contents
+certificate = open("certificate.pem", "rt", encoding="utf-8")
+cert_contents = certificate.read()
+
+# Create x509 enrollment group with an intermediate cert
+client.enrollment_groups.create_or_update(
+    id="<enrollment_group_id>",
+    enrollment_group={
+        "enrollmentGroupId": "<enrollment_group_id>",
+        "attestation": {
+            "type": "x509",
+            "x509": {
+                "signingCertificates": {
+                    "primary": {"certificate": f"{cert_contents}"},
+                    "secondary": {"certificate": f"{cert_contents}"},
+                }
+            },
+        },
+    }
+)
+```
+
+### Create an x509 CA certificate enrollment group
+Create an enrollment group with an x509 CA certificate attestation. 
+This will ensure a registered device's certificate chain has been signed by the target CA cert at the control plane layer.
+```python
+from azure.iot.deviceprovisioning import DeviceProvisioningClient
+
+# Initialize client
+client = DeviceProvisioningClient.from_connection_string(connection_string="<connection_string>")
+
+# Load certificate contents
+ca_certificate = open("ca_certificate.pem", "rt", encoding="utf-8")
+ca_contents = certificate.read()
+
+# Create x509 enrollment group with CA References
+client.enrollment_groups.create_or_update(
+    id="<enrollment_group_id>",
+    enrollment_group={
+        "enrollmentGroupId": "<enrollment_group_id>",
+        "attestation": {
+            "type": "x509",
+            "x509": {
+                "caReferences": {
+                    "primary": f"{ca_contents}",
+                    "secondary": f"{ca_contents}",
+                }
+            },
+        },
+    }
+)
+```
+
+### Check device registration status
+```python
+from azure.iot.deviceprovisioningservice import DeviceProvisioningClient
+
+# Initialize client
+client = DeviceProvisioningClient.from_connection_string(connection_string="<connection_string>")
+
+# Query device registrations for an enrollment group
+device_registrations = client.device_registration_state.query(
+    id="<enrollment_group_id>"
+)
+
+# Get device registration status for a particular device
+state = client.device_registration_state.get(
+    id="<device_id>"
+)
+```
+
+
+## Troubleshooting
```

Comment:
very empty troubleshooting. I would expect the most common issues being authentication errors and/or I tried to add my device and I don't see the registration.